### PR TITLE
[BISERVER-14442] - Database connections using connection pooling will not adhere to changes in the connection details

### PR DESCRIPTION
**Pull request by bcostahitachivantara**
@pentaho-lmartins 
@ssamora 
@ppatricio 
@smmribeiro

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 6h 19s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **623**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7535/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6h 16s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **623**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7535/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/di/core/database/DatabaseMeta.java [hunk 1]
```diff
@@ -913,8 +915,17 @@ public void setIndexTablespace( String index_tablespace ) {
     databaseInterface.setIndexTablespace( index_tablespace );
   }
 
+  boolean isNeedUpdate() {
```
Comment (ppatricio) [inline core/src/main/java/org/pentaho/di/core/database/DatabaseMeta.java:918]: Please change this to a more clearly name: isUpdateNeeded()

**Comment by buildguy** [comment]
### :x: Build failed in 6h 14s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **623**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7535/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 3m 19s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **623**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7535/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :warning: Build finished in 30m 18s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 20 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testJsonInputExternalResourceConsumer</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testGetUsedFields_isNotAcceptingFilenames</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testCloneAnalyzer</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testGetUsedFields_fileNameFromField</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testGetSupportedSteps</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testIsOutput</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testGetResourceOutputNodeType</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testNewInstance</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testGetResourceInputNodeType</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testIsInput</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testCloneAnalyzer</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testJsonOutputExternalResourceConsumer</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testGetSupportedSteps</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testIsOutput</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testGetResourceOutputNodeType</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testNewInstance</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testGetResourceInputNodeType</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testIsInput</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.getxmldata.GetXMLDataStepAnalyzerTest.testCreateResourceNode</b> (click to expand)</summary><p>

```
Could not initialize class org.pentaho.metaverse.api.analyzer.kettle.KettleAnalyzerUtil
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.xmloutput.XMLOutputStepAnalyzerTest.testCreateResourceNode</b> (click to expand)</summary><p>

```
org/osgi/framework/ServiceFactory
```

</p></details>

  
Tests run: **4877**, Failures: **20**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7535/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :warning: Build finished in 23m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 20 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testJsonInputExternalResourceConsumer</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testGetUsedFields_isNotAcceptingFilenames</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testCloneAnalyzer</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testGetUsedFields_fileNameFromField</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testGetSupportedSteps</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testIsOutput</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testGetResourceOutputNodeType</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testNewInstance</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testGetResourceInputNodeType</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsoninput.analyzer.JsonInputAnalyzerTest.testIsInput</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testCloneAnalyzer</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testJsonOutputExternalResourceConsumer</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testGetSupportedSteps</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testIsOutput</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testGetResourceOutputNodeType</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testNewInstance</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testGetResourceInputNodeType</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.jsonoutput.analyzer.JsonOutputAnalyzerTest.testIsInput</b> (click to expand)</summary><p>

```
org/springframework/security/core/Authentication
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.getxmldata.GetXMLDataStepAnalyzerTest.testCreateResourceNode</b> (click to expand)</summary><p>

```
Could not initialize class org.pentaho.metaverse.api.analyzer.kettle.KettleAnalyzerUtil
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.xmloutput.XMLOutputStepAnalyzerTest.testCreateResourceNode</b> (click to expand)</summary><p>

```
org/osgi/framework/ServiceFactory
```

</p></details>

  
Tests run: **4877**, Failures: **20**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7535/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 26m 31s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **4886**, Failures: **0**, Skipped: **7** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7535/7/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**